<a href="https://colab.research.google.com/github/Tanu-N-Prabhu/Python/blob/master/Learn_the_Python_Math_Module.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Learn the Python Math Module

In this tutorial, I will show you how to use to the builtin Python `math` module.

![alt text](https://miro.medium.com/max/743/1*EfK0rFMPhh57Q70auf8T_w.png)

# What is the math module?


It is the C standard defined math module. The math module provides access to mathematical functions.

I know that I have used some jargon in the above statement. Let me clear the clutter. The important terms in the above statement are ***module, C standard, mathematical functions***.



*   **Module**: A file containing a set of **statements** and **definitions** is called a module. For example, the file `program.py` is called a module and the module name would be `program`. You will know how to use a module in the later tutorial.

*   **C standard**: We also call it `libc` (which means library C), which is the standard library of the C programming language. The C standard library provides several string handling functions, data types, mathematical computations, process control, and many more.

*   **Mathematical functions**: These are a group of functions provided by the standard C library. They include `abs`, `labs`, `div`, `max`, `min`, and many more. To read more about the mathematical functions, there’s nothing better than [Wikipedia](https://en.wikipedia.org/wiki/C_mathematical_functions).




---




# Why do we need to use the math module?


Well, the answer is simple: Whenever we need to perform mathematical operations then the math module comes in handy. Let me give you a simple example. It’s quiz time!


## Calculate the area of a regular python with 4 sides and length of a side is 20

Let give me you some parameters and the formulae of the area. Now just calculate the area without using the math module in python. Pretty challenging right?

**The formulae to find the area of a regular polygon is:**


![alt text](https://miro.medium.com/max/1174/1*PbzNnTSCatkptevYEhja1g.png)


**Now the parameters are:**

*  n = number of sides
*  l = length of a side

Now you have things clarified, give me the answer (the program to solve it without using the math module) in about 5–7 minutes. You can post the answer in the comment section below.


![alt text](https://miro.medium.com/max/2115/1*7DBF85m70NNuujpg7uEBMA.png)


## Solution

I know most of you guys might end up calculating the values using a calculator and then put that in the formulae and then display the result programmatically. It will get the correct solution, but it is definitely not recommended.


**Let me give you the program first (using math module) and then I will explain it line by line**





In [1]:
import math
area = (4 * math.pow(20, 2))/(4 * math.tan(math.pi/4))
print(area)

400.00000000000006




*   The first step is to import the math module. This can be done as `import math`. This is what gives you access to using its mathematical methods.

*   The second step is just substituting the values to the formulae. Now here is the significance of the math module. I have used three important methods of the math module and those are `math.pow()`, `math.tan()`, `math.pi`.

The `math.pow()` → It is a built-in power function that is used to calculate the exponent of the given number used in the math module.


```
pow(...)     
    pow(x, y)          
    Return x**y (x to the power of y).
```

## Calculating 150 to the power 4




In [2]:
import math
print(math.pow(150, 4))

506250000.0


The `math.tan()` → It is used to calculate the tangent of a given number in radians.




```
tan(...)     
    tan(x)          
    Return the tangent of x (measured in radians).
```



## Calculate the tangent of 60


In [3]:
import math
print(math.tan(60))

0.320040389379563


The `math.pi` → As the name suggests is used as the constant value of Pi (3.14159265359). It returns the floating value of pi, rather than memorizing and then writing the value of pi, just use the math module.


## Print the value of Pi



In [4]:
import math
print(math.pi)

3.141592653589793


Back to the explanation.


* The third step is just printing the result, in this case, the area is the final result. 400.00000000000006 is the area of the regular polygon.


Now that you guys have mastered the math module, I need you guys to solve the question and let me know the answer below in the comment section. Whoever gets this answer correct will get a **brofist** from me.



---


# Question

## Calculate the area of the regular hexagon which has 6 sides.

**Hint**: a = 6, also for more details on regular hexagon visit [here](https://byjus.com/maths/area-of-hexagon-formula/)



---

Also, there are exactly 44 functions in the math module. How do I know? Well I literally counted every one of them using the help function

`print(help(math))`


The above code on executing displays all the mathematical functions in the math module. There would be no reason for me to explain every single function. Just remember one thing, just use the resources only when you need them. There is no need for you need to memorize all the functions. If you are using an [interactive python notebook](https://jupyter.org/), then you can check the syntax of function without even knowing them as shown below:


![alt text](https://miro.medium.com/max/1382/1*fYk39dmy0XurELPU-1n_Fw.png)

But if you are a geek and you love to read the official documentation of the math module and want to know the syntax of every single function, then there is no better than the [Python math module documentation](https://docs.python.org/3/library/math.html).

Anyway, I hope that you guys learned something new today. Learning is the stepping stone of success. So keep learning. Stay tuned for more updates.


![alt text](https://miro.medium.com/max/1280/1*0VoACtfamg5rUQyfWD8OhA.jpeg)




---


